src/index.ts

```diff
@@ -136,21 +137,22 @@ const attempt = async <T, E extends Error>(
 
 export const go = async <T, E extends Error>(
   fn: () => Promise<T>,
-  options?: GoAsyncOptions
+  options?: GoAsyncOptions<E>
 ): Promise<GoResult<T, E>> => {
   if (!options) return attempt(fn);
 
-  const { retries, attemptTimeoutMs, delay, totalTimeoutMs } = options;
+  const { retries, attemptTimeoutMs, delay, totalTimeoutMs, onAttemptError } = options;
 
-  let fullTimeoutExceeded = false;
+  let fullTimeoutExceededGoResult: GoResult<never, Error> | null = null;
   let totalTimeoutCancellable: CancellableTimeout | null = null;
   let fullTimeoutPromise = new Promise((_resolve) => {}); // Never resolves
   if (totalTimeoutMs !== undefined) {
     // Start a "full" timeout that will stop all retries after it is exceeded
     totalTimeoutCancellable = cancellableSleep(totalTimeoutMs);
     fullTimeoutPromise = totalTimeoutCancellable.promise.then(() => {
-      fullTimeoutExceeded = true;
-      return fail(new Error('Full timeout exceeded'));
+      const goRes = fail(new Error('Full timeout exceeded'));
+      fullTimeoutExceededGoResult = goRes;
+      return goRes;
```

Comment:
I think this can be reverted back to just a boolean flag. At least I think the `fullTimeoutExceededGoResult` is actually never used. I mean the value, not just its existence.

Reply:
Yeah, it could. I was just lazy to update it back.